File: datasets_v2/communication/ChatSecureAndroid/compose/add_contact_activity.kt
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.paint
import androidx.compose.ui.graphics.Color
import androidx.compose.foundation.Image
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.sp
import androidx.compose.ui.unit.dp

@Composable
fun add_contact_activity() {
  Column(modifier = Modifier.background(Color.Unspecified).fillMaxWidth().fillMaxHeight().padding(6.0.dp)) {
    Text(text = "Account to add to:", modifier = Modifier.fillMaxWidth().wrapContentHeight())
    val spinnerOptions_choose_list = listOf("Account to add to:", "Option 1", "Option 2", "Option 3")
    var spinnerExpanded_choose_list by remember { mutableStateOf(false) }
    var spinnerSelected_choose_list by remember { mutableStateOf(spinnerOptions_choose_list.first()) }
    Box(modifier = Modifier.fillMaxWidth().wrapContentHeight()) {
      OutlinedButton(onClick = { spinnerExpanded_choose_list = true }, modifier = Modifier.fillMaxWidth()) {
        Text(text = spinnerSelected_choose_list)
      }
      DropdownMenu(expanded = spinnerExpanded_choose_list, onDismissRequest = { spinnerExpanded_choose_list = false }) {
        spinnerOptions_choose_list.forEach { option ->
          DropdownMenuItem(
            text = { Text(option) },
            onClick = {
              spinnerSelected_choose_list = option
              spinnerExpanded_choose_list = false
            }
          )
        }
      }
    }
    Text(text = "Username or JabberID of person to add:", modifier = Modifier.fillMaxWidth().wrapContentHeight())
    /* AutoComplete requires data source (adapter) */
    OutlinedTextField(value = "", onValueChange = {}, modifier = Modifier.fillMaxWidth().wrapContentHeight())
    Button(onClick = {}, enabled = true, modifier = Modifier.wrapContentWidth().wrapContentHeight()) {
      Text(text = "Send Invite", fontSize = 24.sp)
    }
    Button(onClick = {}, enabled = true, modifier = Modifier.wrapContentWidth().wrapContentHeight()) {
      Text(text = "Scan QR", fontSize = 24.sp)
    }
  }
}


@Preview(showBackground = true)
@Composable
fun add_contact_activity_preview() {
    MyApplicationTheme {
        add_contact_activity()
    }
}

File: experiments/communication/ChatSecureAndroid/swiftui/Add_contact_activity.swift
import SwiftUI

struct Add_contact_activity: View {
  var body: some View {
    VStack(alignment: .leading, spacing: 0) {
      Text("Account to add to:").frame(maxWidth: .infinity, alignment: .leading)
      Picker("", selection: .constant(0)) {
        Text("Account to add to:").tag(0)
        Text("Option 1").tag(1)
        Text("Option 2").tag(2)
        Text("Option 3").tag(3)
      }
      .pickerStyle(.menu)
      .frame(maxWidth: .infinity)
      Text("Username or JabberID of person to add:").frame(maxWidth: .infinity, alignment: .leading)
      /* AutoComplete requires data source (adapter) */
      TextField("", text: .constant("")).frame(maxWidth: .infinity)
      Button(action: { }) {
        Text("Send Invite").font(.system(size: 24))
      }
      .disabled(false)
      Button(action: { }) {
        Text("Scan QR").font(.system(size: 24))
      }
      .disabled(false)
    }
    .background(Color.clear)
    .frame(maxWidth: .infinity, maxHeight: .infinity)
    .padding(6.0)
  }
}

struct Add_contact_activity_Previews: PreviewProvider {
  static var previews: some View {
    Add_contact_activity()
  }
}
